conda_env/cli/main_create.py

```diff
@@ -22,6 +23,7 @@
     conda env create -n=foo
     conda env create -f=/path/to/environment.yml
     conda env create --name=foo --file=environment.yml
+    conda env create --name=foo --binstar=username/foo
```

Comment:
They should conform to an API like this (doesn't have to be this specifically, but this gives you an idea):

```python
class EnvironmentLoader(object):
    def can_handle(self, name):
        # return True if you can process, False if you can't

    def load(self, name):
        # return an Environment object
```

The main thing is that a processor needs to be able to say that it can deal with a request and then actually deal with the request.

So from there, imagine a Binstar one that might look like this (Github would look similar):

```python
class BinstarLoader(EnvironmentLoader):
    def can_handle(self, name):
        if "/" not in name:
            return False  # Bail early
        # attempt to load from binstar, return True if its found
    
    def load(self, name):
        # fetch environment.yml from binstar
        return env.Environment(url=binstar_url, **data_loaded_from_binstar)
```

This should be done by loading a list of backends (look at the [get_installer][1] for examples) that are possibly able process the backend.  This is a good case for `pkg_resources.iter_entry_points`.  Now that we install `pip` by default in every environment that `python` is installed there shouldn't be any resistance to using it (see [this diff][2] for examples for loading entry points).

[1]: https://github.com/conda/conda-env/blob/develop/conda_env/installers/base.py#L11-L15
[2]: https://github.com/conda/conda-env/pull/64/files#diff-a6b0af01b802da645b8d8a37879fa3c6R40